### PR TITLE
Refactor `storage_dataflow_max_inflight_bytes_config`

### PR DESCRIPTION
**Pull request by moulimukherjee**
Most of the changes are a revert of https://github.com/MaterializeInc/materialize/pull/20688.

Instead of passing the clsuter sizes over to the upsert source, currently sending the memory limit if defined to clusterd which will eventually be available in storage state.

### Motivation
Fixes https://github.com/MaterializeInc/materialize/issues/20736

### Tips for reviewer
Main changes are at

- https://github.com/MaterializeInc/materialize/pull/20752/files#diff-a7392de31a3a6faa1ece80ff79d07be99199704c6f6787546ab0c4d794129381R502-R509
- https://github.com/MaterializeInc/materialize/pull/20752/files#diff-e8e7455b8c7dd550a16ee31db4ba293ed3ab1cced5505cf905bab8c3774c0d0aR329

Majority of the other changes are a revert for the cluster_size plumbing.

### Checklist

- [x] This PR has adequate test coverage / QA involvement has been duly considered.
- [x] This PR has an associated up-to-date [design doc](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/README.md), is a design doc ([template](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/00000000_template.md)), or is sufficiently small to not require a design.
  
- [ ] If this PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way), then it is tagged with a `T-proto` label.
- [ ] If this PR will require changes to cloud orchestration, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [ ] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):
  -

### CODE REVIEW

File: test/upsert/rehydration/01-setup.td [hunk 1]
```diff
@@ -9,12 +9,6 @@
...
-      STORAGECTL ADDRESSES ['clusterd1:2100'],
-      STORAGE ADDRESSES ['clusterd1:2103'],
-      COMPUTECTL ADDRESSES ['clusterd1:2101'],
-      COMPUTE ADDRESSES ['clusterd1:2102'],
-      WORKERS 4
-    )
-  )
+# Using mem-2 because it has defined memory limits which will be passed for
+# backpressure inflight bytes configuration
+> CREATE CLUSTER storage_cluster SIZE 'mem-2', REPLICATION FACTOR 1
```
Comment (moulimukherjee) [inline test/upsert/rehydration/01-setup.td:14]: Is this fine? Otherwise, can also pass the memory limit directly to clusterd in mzcompose.

Comment (benesch) [reply test/upsert/rehydration/01-setup.td:14]: I'm not sure myself if it was important for this test to be using the existing `clusterd1` container!

**Review by benesch — APPROVED** [review]
This looks fantastic! Thank you so much for doing this.

File: test/upsert/rehydration/01-setup.td (hunk 1 above)
Comment (benesch) [reply test/upsert/rehydration/01-setup.td:14]: Probably need @guswynn to weigh in on this one.

**Review by teskje — APPROVED** [review]
Compute changes LGTM. Thanks!

File: test/upsert/rehydration/01-setup.td (hunk 1 above)
Comment (guswynn) [reply test/upsert/rehydration/01-setup.td:14]: yeah this has to be the existing clusterd1 container

Comment (moulimukherjee) [reply test/upsert/rehydration/01-setup.td]: updated! passing the memory limit via clusterd args instead

**Pull request merged**